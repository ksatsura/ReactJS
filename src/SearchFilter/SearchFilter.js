//imports
import React from 'react';
import FilterButton from '../FilterButton/FilterButton';
import { connect } from 'react-redux';
import { fetchAssetsIfNeeded } from '../actions/searchActions';

export const SearchFilter = (props) => {

    const { value, handleSearchBtnClick } = props;

    return (
        <div className='search-buttons'>
            <div className='search-filter'>
                <p className='search-filter-title'>search by</p>
                <FilterButton id='title' name='title' />
                <FilterButton id='genre' name='genre' />
            </div>
            <input type="submit" value="Search" className='search-button' onClick={() => handleSearchBtnClick(value) }/>
        </div>
    );
};

const mapDispatchToProps = dispatch => {
    return {
        handleSearchBtnClick: (value) => dispatch(fetchAssetsIfNeeded(value)),
    };
};

const mapStateToProps = state => {
    return {
        value: state.filmListReducer.currentValue,
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchFilter);