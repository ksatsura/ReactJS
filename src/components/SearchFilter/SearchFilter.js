//imports
import * as React from 'react';
import FilterButton from '../FilterButton/FilterButton';
import { connect } from 'react-redux';
import { fetchRequestIfNeeded } from '../../redux-utils/asyncActionUtils';
import { Link }  from 'react-router-dom';
import '../../style.css';

// @flow
type Props = {
    paramValue: string, 
    handleUrlChange: (string) => void,
    value: string,
};

export class SearchFilter extends React.Component<Props> {

    componentDidUpdate(prevProps) {
        if (this.props.paramValue !== prevProps.paramValue) {
            this.props.handleUrlChange(this.props.paramValue);
        }
    }

    componentDidMount() {
        this.props.paramValue && this.props.handleUrlChange(this.props.paramValue);
    }

    render() {
        return (
            <div className='search-buttons'>
                <div className='search-filter'>
                    <p className='search-filter-title'>search by</p>
                    <FilterButton id='title' name='title' />
                    <FilterButton id='genre' name='genre' />
                </div>
                <Link to={`/search/${this.props.value}`}>
                    <input type="submit" value="Search" className='search-button' />
                </Link>
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        handleUrlChange: (value) => dispatch(fetchRequestIfNeeded(value)),
    };
};

const mapStateToProps = state => {
    return {
        value: state.filmListReducer.currentValue,

    };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchFilter);