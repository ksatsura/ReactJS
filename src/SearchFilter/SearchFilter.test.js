import React from 'react';
import { SearchFilter } from './SearchFilter';
import { shallow } from 'enzyme';

describe('SearchFilter component', () => {

    const props = {
        value: ' ',
        handleSearchBtnClick: jest.fn(),
    };

    describe('renders', () => {

        it('renders for desktop', () => {
            const component = shallow(<SearchFilter {...props} />);

            expect(component).toMatchSnapshot();
        });

        it('renders for mobile', () => {
            const component = shallow(<SearchFilter isMobile {...props} />);

            expect(component).toMatchSnapshot();
        });
    });

    describe('handleChange', () => {

        it('test input change event', () => {
            
            const component = shallow(<SearchFilter {...props} />);

            component.find('input').simulate('click');
            expect(props.handleSearchBtnClick.mock.calls.length).toEqual(1);
        });
    });
});
