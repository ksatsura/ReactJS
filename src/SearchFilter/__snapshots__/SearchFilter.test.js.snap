// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter component renders renders for desktop 1`] = `ShallowWrapper {}`;

exports[`SearchFilter component renders renders for mobile 1`] = `ShallowWrapper {}`;
